File: Day1/part2.py
#!/usr/bin/env python3

from functools import reduce 

with open('input', 'r') as f:
    deers = [] # plural of deer blah blah blah
    deer  = 0
    for l in f:
        l = l.rstrip()
        if len(l):
            deer = deer + int(l);
        else:
            deers.append(deer)
            deer = 0
    deers.sort(reverse=True)
    print(reduce(lambda a,b: a+b, deers[:3])) 
    

File: Day1/part2.js
#!/usr/bin/env node

let deers = [];
let deer  = 0;
require('fs').readFileSync('input', 'utf-8').split(/\r?\n/).forEach((l) => {
    if (! l.length) {
        deers.push(deer);
        deer = 0;
    }
    else deer += parseInt(l, 10);
});
console.log(deers.sort((a, b) => b-a).slice(0, 3).reduce((a, b) => a+b));